### PR TITLE
Allow different timestamp formats in Google exports to be parsed

### PR DESCRIPTION
**Pull request by saschazepter**
Older exports may have timestamps in millisecond format, i. e. `startTimestampMs` instead of `startTimestamp` and `timestampMs` instead of `timestamp`. This might make the import of a `Location History.json` or `Records.json` fail or leaving back zero imports at all.

The changed files will help these files getting parsed correctly.

### CODE REVIEW

File: app/services/google_maps/records_parser.rb [hunk 1]
```diff
@@ -35,10 +35,25 @@ def parse_json(json)
...
       longitude: json['longitudeE7'].to_f / 10**7,
-      timestamp: DateTime.parse(json['timestamp']).to_i,
+      timestamp: parse_timestamp(json['timestamp'] || json['timestampMs']),
       altitude: json['altitude'],
       velocity: json['velocity'],
       raw_data: json
     }
   end
+
+  def parse_timestamp(timestamp)
```
Comment (Freika) [inline app/services/google_maps/records_parser.rb]: This method is the same across `app/services/google_maps/records_parser.rb` and `app/services/google_maps/semantic_history_parser.rb`. Can you please move it to a module so it wouldn't be repeated?

Also tests for both cases (with `timestamp` and `timestampMs`) would be great to have.

**Review by Freika — CHANGES_REQUESTED** [review]
Thank you for taking care of this issue! Although, there is some changes to be made.

File: app/services/google_maps/records_parser.rb (hunk 1 above)
Comment (Freika) [reply app/services/google_maps/records_parser.rb]: Master branch now has tests for both classes changed in this PR, so you can take them as an example

Comment (saschazepter) [reply app/services/google_maps/records_parser.rb]: Where would you move the ```parse_timestamp``` method? i would place it in the ```application_helper.rb```.

Comment (Freika) [reply app/services/google_maps/records_parser.rb]: Code defined in helpers by default is only available in views, not in service classes. I'd place it in module somewhere in /lib/timestamps.rb, for example

**Comment by Freika** [comment]
@saschazepter is it ready for review?

**Comment by saschazepter** [comment]
> @saschazepter is it ready for review?

Not with the tests yet.

**Review by Freika — APPROVED** [review]
Great job, thank you for fixing it!

**Pull request merged**